webrender_api/src/image.rs

```diff
@@ -222,44 +298,64 @@ pub trait BlobImageHandler: Send {
 
 /// A group of rasterization requests to execute synchronously on the scene builder thread.
 pub trait AsyncBlobImageRasterizer : Send {
+    /// Rasterize the requests.
     fn rasterize(&mut self, requests: &[BlobImageParams]) -> Vec<(BlobImageRequest, BlobImageResult)>;
 }
 
 
+/// FIXME: document
 #[derive(Copy, Clone, Debug)]
 pub struct BlobImageParams {
+    /// FIXME: document
     pub request: BlobImageRequest,
+    /// FIXME: document
     pub descriptor: BlobImageDescriptor,
+    /// FIXME: document
     pub dirty_rect: Option<DeviceUintRect>,
 }
 
+/// Backing store for blob image command streams.
 pub type BlobImageData = Vec<u8>;
 
+/// Result type for blob raserization.
 pub type BlobImageResult = Result<RasterizedBlobImage, BlobImageError>;
 
+/// Metadata (but not storage) for a blob image.
 #[repr(C)]
 #[derive(Copy, Clone, Debug)]
 pub struct BlobImageDescriptor {
+    /// Bounding rectangle of the image, in device pixels.
     pub size: DeviceUintSize,
+    /// FIXME - Document me.
     pub offset: DevicePoint,
+    /// Format for the data in the backing store.
     pub format: ImageFormat,
 }
 
+/// Representation of a rasterized blob image. This is obtained by passing
+/// `BlobImageData` to the embedding via the rasterization callback.
 pub struct RasterizedBlobImage {
+    /// The bounding rectangle for this bob image.
     pub rasterized_rect: DeviceUintRect,
+    /// Backing store. The format is stored out of band in `BlobImageDescriptor`.
     pub data: Arc<Vec<u8>>,
 }
 
+/// Error code for when blob rasterization failed.
 #[derive(Clone, Debug)]
 pub enum BlobImageError {
+    /// Out of memory.
     Oom,
-    InvalidKey,
-    InvalidData,
+    /// Other failure, embedding-specified.
     Other(String),
 }
 
+/// FIXME: document
 #[derive(Copy, Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
 pub struct BlobImageRequest {
+    /// Unique handle to the image.
     pub key: ImageKey,
+    /// Tiling offset.
+    /// FIXME - document tiling better.
```

Comment:
/// Tile offset in number of tiles, if applicable.
///
/// If set to `None` it means the image is not tiled.